transport-native-unix-common/src/main/java/io/netty/channel/unix/UnixChannelUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.unix;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.internal.PlatformDependent;
+
+import static io.netty.channel.unix.Limits.IOV_MAX;
+
+public final class UnixChannelUtil {
+
+    private UnixChannelUtil() {
+    }
+
+    /**
+     * Checks if the specified buffer has memory address or is composed of n(n <= IOV_MAX) NIO direct buffers.
+     * (We check this because otherwise we need to make it a new direct buffer.)
+     */
+    public static boolean isBufferNoNeedToConvert(ByteBuf byteBuf) {
+        if (byteBuf.hasMemoryAddress()) {
```

Comment:
Also we shouldn't have to check `PlatformDependent.hasUnsafe()` ... this is currently a requirement for the native transports.